api/providers/AddressProvider.cs

```diff
@@ -53,4 +54,23 @@ public List<Address> GetAll()
     {
         throw new NotImplementedException();
     }
+
+    public Address? GetOrCreateAddress(SupplierRequest request)
+    {
+
+        if (request.Address_id != null)
+        {
+            Address? existingAddress = GetById(request.Address_id.Value);
```

Comment:
```suggestion
            Address? existingAddress = GetById(request.AddressId.Value);
```

Graag PascalCase aanhouden. (en vranderen in de addressDTO)